File: SimpleEngine/Mesh.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SimpleEngine
{
    public class Mesh : Media
    {
        CompiledMesh mesh = null;

        public override void load(MediaLoader ml, FileSystem fs, string path)
        {
            MeshDef mesh = MeshFile.Load(fs, path);
            mesh.compile(ml);
            this.mesh = mesh.Compiled;
        }

        protected override void unload()
        {
            mesh = null;
        }

        public CompiledMesh Compiled
        {
            get
            {
                return mesh;
            }
        }
    }
}

File: src/engine/mesh.cc
#include "engine/mesh.h"

#include <memory>

#include "engine/compiledmesh.h"
#include "engine/media.h"
#include "engine/meshfile.h"

namespace simple
{
    void Mesh::Load(MediaLoader* ml, FileSystem* fs, const std::string& path)
    {
        auto mesh = mesh_file::Load(fs, path);
        mesh->Compile(ml);
        this->mesh = mesh->GetCompiledMesh();
    }
}
